packages/react-components/react-tree/src/components/Tree/useTree.ts

```diff
@@ -119,7 +118,7 @@ function useRootTree(props: TreeProps, ref: React.Ref<HTMLElement>): TreeState {
         const element = targetDocument.getElementById(groupId);
         if (treeWalker && element) {
           treeWalker.currentNode = element;
-          const firstTreeItem = treeWalker.firstChild() as TreeItemElement | null;
+          const firstTreeItem = treeWalker.firstChild() as HTMLElement | null;
```

Comment:
Instead of casting I would create type predicates

```ts
function isHTMLElement(element: Element): element is HTMLElement {
  return element instanceof HTMLElement;
}
```

this probably makes sense in `react-utilities`

Reply:
wouldn't that fail on some cases if we don't use the proper document? @ling1726

Reply:
yh true I think @bsunderhus  made a better check than `instanceof` we should dig that up and make it a shared utility. We are generally moving more towards DOM with react wrapping so it makes sense to have

Reply:
@petr-duda  hopefully this makes things clearer https://www.typescriptlang.org/play?#code/MYewdgzgLgBAtgSwOYAsoDkRQEIFMASAKgLIAyAogDa5y5hQBcMVNdsAvDACYjACutegDpgAJ1wBDKLhaCoACgDkXBADdFASgDcAWABQ+gPSGYhFAggwA7gkqUYuUaJCiYAI1zAJfCLhgJYRFRYMCx3PwkYIjJZNn1QSFguJmiKajkYTiC0TBwCEjTWel0DPWMYAHU-HwiYKABPAAc-RvEVL2k6kBgJCF9RWCgUKX9YCx6ogtj6fQQAMxh5C1TphWyMLDwV9LYNDRgAb30YGAToGDmUqZ36TPhkHM38mJuoEoBffX05vjBgKAQ4H8EG2RQUuEoTFWGiYEOBkxeYMOxxg5UIAHkACLo-yJCR-XAgBY8XCWUKwKwuADWp2cfWsCDAPCsABoLoyuBMPFBpK5gChPFSUeU0FBGhAGMYuLhVBCQM1REI4CAAF62SgSIQuJCGOgAWgAqgBlQw8YAQQxVNyGABSElUEiNYgQjSghgASrg5o46MBcIZ0QqpC4LYzoPi-USAMRhqARwlzAD6+K4ibgfEoAMa1ET4gklDgEBR80WcNj8aJCMKcn2Rz0JxO4igfFEYDqoj4uBKJ0+pUbuGbrYu+d8HyAA

Reply:
> yh true I think @bsunderhus made a better check than `instanceof` we should dig that up and make it a shared utility. We are generally moving more towards DOM with react wrapping so it makes sense to have

I remember that, but I believe it never got merged (as it was used by the native mechanism on `Dialog` and since we decided to not support native, the verification method got left out of it.